firewall.go

```diff
@@ -278,6 +279,13 @@ func (f *Firewall) GetRuleHash() string {
 	return hex.EncodeToString(sum[:])
 }
 
+// GetRuleHashFNV returns a float64 FNV-1 hash representation the rules, suitable for use as a Prometheus value
+func (f *Firewall) GetRuleHashFNV() float64 {
+	h := fnv.New32a()
```

Comment:
Metrics are `int64` so we should use `fnv.New64a` and return a `int64` instead of a `float64`, will help avoid the already unlikely possibility of a colliding hash.

Reply:
Just so I'm clear - your particular implementation is using int64 for the metric values?  Because the Prometheus data model is float64.

Reply:
Yep, https://pkg.go.dev/github.com/rcrowley/go-metrics#Gauge

Reply:
n/m, I see that go-metrics does indeed use int64 - https://pkg.go.dev/github.com/rcrowley/go-metrics?utm_source=godoc#Gauge

Reply:
The go-metrics integration with prometheus does convert it back to a float64 though. So if we generate an int64 it will be lossy: https://github.com/deathowl/go-metrics-prometheus/blob/f2a1482ba35b474b8a55c052f7333c614b93d36b/prometheusmetrics.go#L156-L157

I think the best bet would be to go back to fnv.New32a, treat it internally as a `uint32`, but the gauge can cast it to `int64` (and then it won't be lossy when prom integration converts it to a float64, since a uint32 should fit in there safely).

Reply:
Yep, agreed and that is super frustrating. Sorry for the run around.

Reply:
No problem - back to New32a and using it as a uint32 internally.